### PR TITLE
feat : JRD Email functionality to test(RDCC 1055 )

### PR DESCRIPTION
**Pull request by choudsc**
Before creating a pull request make sure that:

JIRA link (if applicable)
https://tools.hmcts.net/jira/browse/RDCC-1055

Change description
feat : JRD Email functionality to  test(RDCC 1055 )

Does this PR introduce a breaking change? (check one with "x")

NO

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/juddata/camel/service/EmailData.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package uk.gov.hmcts.reform.juddata.camel.service;
+
+import lombok.Builder;
+import lombok.Getter;
+
+@Getter
+@Builder(toBuilder = true)
```
Comment (abhijitdiwan) [inline src/main/java/uk/gov/hmcts/reform/juddata/camel/service/EmailData.java:7]: do you want to create copies or near-copies of objects? toBuilder is required?

File: src/main/java/uk/gov/hmcts/reform/juddata/camel/service/EmailService.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+            SimpleMailMessage message = new SimpleMailMessage();
+            message.setTo(emailData.recipient);
+            message.setSubject(emailData.subject);
+            message.setText(emailData.message);
+            message.setFrom(from);
+
+            mailSender.send(message);
+
+        } catch (MailException e) {
+            throw new RuntimeException(e);
```
Comment (abhijitdiwan) [inline src/main/java/uk/gov/hmcts/reform/juddata/camel/service/EmailService.java:30]: We need to change this RuntimeException with our own custom. Not now but in Intragtion.

File: src/main/resources/application.yaml [hunk 3]
```diff
@@ -60,8 +73,8 @@ spring:
 
 # Scheduler configuration
```
Comment (abhijitdiwan) [inline src/main/resources/application.yaml:74]: Don't commit this file...for your testing fine.Once testing done revert.

File: src/main/java/uk/gov/hmcts/reform/juddata/scheduler/JrdUserProfileDataLoadScheduler.java [hunk 4]
```diff
@@ -31,7 +28,7 @@
     @PostConstruct
```
Comment (abhijitdiwan) [inline src/main/java/uk/gov/hmcts/reform/juddata/scheduler/JrdUserProfileDataLoadScheduler.java:28]: No change in this class.Revert with master

File: src/integrationTest/java/uk/gov/hmcts/reform/juddata/cameltest/ParentOrchestrationRouteTest.java [hunk 5]
```diff
@@ -182,28 +213,29 @@ public void testParentOrchestrationSingleRecord() throws Exception {
...
+        camelContext.getGlobalOptions().put(Exchange.FAILURE_ROUTE_ID, "Rout 1");
+        ReflectionTestUtils.setField(exceptionProcessor, "mailEnabled", Boolean.FALSE);
+        parentRoute.startRoute();
+        producerTemplate.sendBody(startRoute, "test JRD orchestration");
+        Mockito.verify(emailService, Mockito.times(0)).sendEmail(Mockito.any(String.class), Mockito.any(uk.gov.hmcts.reform.juddata.camel.service.EmailData.class));
     }
 
-    private static void setSourcePath(String path, String propertyPlaceHolder) throws Exception {
+    @Test
+    public void testParentOrchestrationSucessEmail() throws Exception {
```
Comment (abhijitdiwan) [inline src/integrationTest/java/uk/gov/hmcts/reform/juddata/cameltest/ParentOrchestrationRouteTest.java:229]: where is flag cases true and false?

File: src/main/resources/application.yaml [hunk 6]
```diff
@@ -26,6 +26,20 @@ spring:
   jackson.date-format: yyyy-MM-dd
   flyway.enabled: ${FLYWAY_ENABLED:true}
   flyway.locations: classpath:db/migration,classpath:db/postgres
+  mail:
+    from: no-reply@reform.hmcts.net
+    to:  sushant.choudhari@hmcts.net
+    subject:  Exception received at Rout
```
Comment (shashank-rastogi) [inline src/main/resources/application.yaml:32]: same here - subject needs to change

File: src/test/java/uk/gov/hmcts/reform/juddata/camel/email/EmailDataTest.java [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+public class EmailDataTest {
+
+    private static final String EMAIL_TO = "recipient@example.com";
+    private static final String EMAIL_SUBJECT = "Exception received at Rout id";
+
+    private EmailData emailDataUnderTest;
+
+    @Test
+    public void testEmailData() {
+        EmailData emailData = JrdUnitTestHelper.getMockEmail();
```
Comment (shashank-rastogi) [inline src/test/java/uk/gov/hmcts/reform/juddata/camel/email/EmailDataTest.java:18]: JrdUnitTestHelper - name needs to change

File: src/test/java/uk/gov/hmcts/reform/juddata/camel/email/EmailServiceTest.java [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+import org.springframework.mail.SimpleMailMessage;
+import org.springframework.mail.javamail.JavaMailSender;
+import uk.gov.hmcts.reform.juddata.camel.helper.JrdUnitTestHelper;
+import uk.gov.hmcts.reform.juddata.camel.service.EmailData;
+import uk.gov.hmcts.reform.juddata.camel.service.EmailService;
+import uk.gov.hmcts.reform.juddata.exception.EmailFailedSendException;
+
+public class EmailServiceTest {
+    private static final String EMAIL_FROM = "no-reply@exaple.com";
+
```
Comment (shashank-rastogi) [inline src/test/java/uk/gov/hmcts/reform/juddata/camel/email/EmailServiceTest.java:25]: email id has spelling mistake..no-reply@exa**m**ple.com
i

**Pull request closed without merging**